src/components/UserSelect/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import styles from './UserSelect.module.css';
+
+type UserSelectProps = {
+    user?: number,
+    idx: number,
+}
+
+function UserSelect(props: UserSelectProps) {
+    const dispatch = useDispatch();
+    const todos = useSelector((state: {list: { todos: any[] }}) => state.list.todos);
+    React.useEffect(
+        () => {
+            console.log('userSelect');
+            fetch('https://jsonplaceholder.typicode.com/users/').then(
+                (users) => users.json(),
+            ).then(users => setOptions(users))
+        },
+        [],
+    )
+    const [options, setOptions] = React.useState([]);
+
+    const { idx } = props;
+    const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        const changedTodos = todos.map((t, index) => {
+            const res = { ...t }
```

Comment:
1. use full param name for better readability: `todo` instead of `t`.
2. use object destructuring correclty:
```
const changedTodos = todos.map((todo, index) => {
    const { user } = todo;
return index === props.idx ?
  return {
    ...todo,
    user,
  }
: todo;
```